instrumentation/aws/iam-role-cfn-template.yaml

```diff
@@ -24,608 +24,1689 @@ Resources:
+          - Effect: Allow
+            Action:
+              #
+              ##
+              ###
+              #### Serverless Metrics - Kinesis Firehose required S3 Bucket policies
+              ###
+              ##
+              #
+
+              # Grants permission to return the Region that an Amazon S3 bucket resides in.
+              # https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketLocation.html
+              - "s3:GetBucketLocation"
+              # Grants permission to create a new bucket.
+              # https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateBucket.html
+              - "s3:CreateBucket"
+              # Grants permission to delete the bucket named in the URI
+              # https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteBucket.html
+              - "s3:DeleteBucket"
+            Resource:
+              # The bucket name will include unique id to avoid bucket duplication
+              - "arn:aws:s3:::serverless.metrics-firehose-backup-*"
+  #
+  ##
+  ###
+  #### SERVERLESS ROLE ASSUME PERMISSIONS
+  ###
+  ##
+  #
+  PolicyOfServerlessRoleAssumeGroup:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Roles:
+        - !Ref ServerlessRole
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              # Grants permission to pass a role to a service.
+              # https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_passrole.html
+              - "iam:PassRole"
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/ServerlessMetricsRole"
+  #
+  ##
+  ###
+  #### SERVERLESS Metrics ROLE PERMISSIONS
+  ###
+  ##
+  #
+  PolicyOfServerlessMetricsRoleStreamGroup:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      Roles:
+        - !Ref ServerlessMetricsRole
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              #
+              ##
+              ###
+              #### Serverless Metrics Stream - Kinesis Firehose policies
+              ###
+              ##
+              #
+
+              # Grants permission to write a single data record into an Amazon Kinesis Firehose delivery stream.
+              # https://docs.aws.amazon.com/firehose/latest/APIReference/API_PutRecord.html
+              - "firehose:PutRecord"
+              # Grants permission to write multiple data records into a delivery stream in a single call, which can achieve higher throughput per producer than when writing single records.
+              # https://docs.aws.amazon.com/firehose/latest/APIReference/API_PutRecordBatch.html
+              - "firehose:PutRecordBatch"
+            Resource:
+              - !Sub "arn:aws:firehose:*:${AWS::AccountId}:deliverystream/serverless_metrics-firehose"
+  PolicyOfServerlessMetricsRoleFirehoseGroup:
```

Comment:
Why do we need this policy?

Reply:
This is the permissions that needed for the firehose deployed in users AWS account.  This is also the permissions auto added when manually create a firehose through AWS UI.